src/RouteResolver.tsx

```diff
@@ -1,107 +1,150 @@
-import { isEmpty } from 'lodash-es'
+// import docTitleConfig from 'config/docTitleConfig'
 import * as React from 'react'
-import { useLocation } from 'react-router-dom'
-import { SynapseComponents } from 'synapse-react-client'
-import { SynapseComponentWithContext } from 'SynapseComponentWithContext'
-import { GenericRoute, SynapseConfig } from 'types/portal-config'
+import { Route, useLocation, useRouteMatch } from 'react-router-dom'
+import { SynapseComponent } from 'SynapseComponent'
+import { SynapseConfig, GenericRoute } from 'types/portal-config'
 import { scrollToWithOffset } from 'utils'
-import docTitleConfig from './config/docTitleConfig'
-import routesConfig from './config/routesConfig'
-import PortalComponents from './portal-components/'
+import docTitleConfig from 'config/docTitleConfig'
+import routesConfig from 'config/routesConfig'
 import Layout from './portal-components/Layout'
-import sharedRouteConfig, { homeRoute } from './shared-config/sharedRoutes'
+import sharedRouteConfig from './shared-config/sharedRoutes'
 
-/**
- * Given a pathname, return the matching route object and the route's pathname.
- * @param pathname
- * @returns
- */
-export const getRouteFromParams = (
-  pathname: string,
-): [GenericRoute, string] => {
-  // e.g. pathname = /Explore/Programs
-  const split: string[] = pathname.split('/').slice(1)
-  // e.g. split = 'Explore', 'Programs
-  // if the last element is index.html (case insensitive, 'l' optional)
-  if (split[split.length - 1].match(/index\.html?/gim)) {
-    // remove index.html
-    split.pop()
-    if (split.length === 0) {
-      // need to have at least 1 items
-      split.push('')
-    }
-  }
-  let route = ([...sharedRouteConfig, ...routesConfig] as GenericRoute[]).find(
-    (el) => split[0] === el.to,
-  )!
+const ROUTES: GenericRoute = {
+  ...sharedRouteConfig,
+  routes: [...(sharedRouteConfig.routes ?? []), ...routesConfig],
+}
 
-  if (route == null) {
-    console.warn(`Error: url at ${pathname} has no route mapping`)
-    return [homeRoute, '/']
+function ComponentRenderer(props: { config: SynapseConfig }) {
```

Comment:
This just renders the synapseConfigArray. Was essentially just pulled out of the previous implementation with no major changes.